### DISCUSSION TITLE
Dependabot should be able to `bundle update <package>`

### DISCUSSION THREAD

**Discussion by mcandre** [community/community#12314] [Code Security]
Dependabot still stinks at generating security PR's, particularly for Ruby projects. Just run `bundle update <package>` and submit the resulting gem lock.

As a workaround I manually run this command.

**Comment by deivid-rodriguez** [comment]
Can you explain what's wrong with dependabot here? You said it stinks at generating security PR's, but you didn't explain why.

**Comment by mcandre** [comment]
Dependabot is unable to perform basic, highly automatable functions such as
using `bundle update` to produce useful security PR's.

Instead, Dependabot mostly generates a warning about how it was unable to
make a PR.

Seems to me that PR's for Ruby projects should largely be doable. I haven't
read the Dependabot code so I can't say why it's not simply doing `bundle
update`. But on the surface, it feels like an obvious solution.

**Comment by deivid-rodriguez** [comment]
Dependabot creates correct security PRs for Ruby projects all the time. I guess you're hitting some issue, I suggest to report it to the dependabot project so it can be corrected.

**Comment by mcandre** [comment]
Sometimes it does, other times it doesn't.

I wish I knew why.

When Dependabot does generate a PR, I think it does a very good job. I wish
it could successfully complete that operation more of the time. And would
provide more detail when it fails to do so.

On my machine, running `bundle update` usually works around this glitch. My
DevOps experience leads me to suspect some kind of intermittent outage in
the Dependabot pipeline. By simply including more details around the PR
generation failure in the reports, we can get a more complete picture of
why it fails (a significant portion of the time).

**Comment by deivid-rodriguez** [comment]
In my library I also run into a dependabot error recently: https://github.com/activeadmin/activeadmin/security/dependabot/17. Isn't the explanation of the error quite good? There's even a button to view full logs.

**Comment by mcandre** [comment]
I can't follow that link, as I am not a repository owner. But I am glad to
learn about a detailed log link. Next time I get an error, I may read the
logs.
